.eslintrc.js

```diff
@@ -1,22 +1,73 @@
 module.exports = {
   root: true,
-  extends: ["@react-native-community", "plugin:i18next/recommended"],
+  parser: "@babel/eslint-parser",
+  parserOptions: {
+    requireConfigFile: false,
+    babelOptions: {
+      presets: ["@babel/preset-react"]
+    }
+  },
+  extends: ["airbnb", "plugin:i18next/recommended"],
+  globals: {
+    FormData: true
+  },
   rules: {
-    quotes: [2, "double"],
+    "arrow-parens": [2, "as-needed"],
     "comma-dangle": [2, "never"],
-    "space-in-parens": [2, "always"],
-    "prettier/prettier": 0,
+    "consistent-return": [2, { treatUndefinedAsUnspecified: true }],
+    "func-names": 0,
+    "global-require": 0,
     "i18next/no-literal-string": [2, {
       words: {
         // Minor change to the default to disallow all-caps string literals as well
         exclude: ["[0-9!-/:-@[-`{-~]+"]
       }
     }],
-    "no-var": 1
+    // The AirBNB approach at
+    // https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/imports.js#L71
+    // is quite particular and forbids imports of devDependencies anywhere
+    // outside of tests and conventional config files, which causes problem
+    // if you have, say, some utility scripts that have nothing to do with
+    // your actually application. Here I'm trying to use the defaults
+    // (https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md,
+    // which allow imports of anything declared in package.json, but should
+    // raise alarms when you try to import things not declared in
+    // package.json.
+    "import/no-extraneous-dependencies": ["error", {}],
+    "max-len": ["error", 100, 2, {
+      ignoreUrls: true,
+      ignoreComments: false,
+      ignoreRegExpLiterals: true,
+      ignoreStrings: false,
+      ignoreTemplateLiterals: false
+    }],
+    "no-alert": 0,
+    "no-underscore-dangle": 0,
+    "no-unused-vars": ["error", {
+      vars: "all",
+      args: "after-used",
+      // Overriding airbnb to allow leading underscore to indicate unused var
+      argsIgnorePattern: "^_",
+      ignoreRestSiblings: true
+    }],
+    "no-void": 0,
+    "prefer-destructuring": [2, { object: true, array: false }],
```

Comment:
Are you saying something like the following will always fail on Android but not in iOS?

```javascript
const thing = { foo: "bar" };
const { baz } = thing;
```

Is that for all versions of Android, or just certain ones? I would have assumed that kind of es6 syntax would be removed via transpilation before getting run though the device js interpreter.

Reply:
I believe it happens across all versions of Android, but only on particular devices. I've noticed it in Seek crashes from the Play Store like [#293](https://github.com/inaturalist/SeekReactNative/issues/293), where a nested object on a new screen like `const { commonAncestor } = taxon` was undefined. Changing that code to `const commonAncestor = taxon.commonAncestor` wouldn't necessarily fix the underlying issue, but it would stop the app from crashing on Android. There didn't ever seem to be corresponding crashes on iOS.

Reply:
That's pretty weird. From what I can see in that issue I'm not convinced destructuring was really the problem, and it's possible flow checks will prevent that from happening here, so I'd like to wait until we have repro conditions before we blame destructuring. If RN transpiles all the JS to es5, destructuring should get automatically removed before it gets to the device.